src/goDeclaration.ts

```diff
@@ -111,27 +111,13 @@ function definitionLocation_godef(input: GoDefinitionInput, token: vscode.Cancel
 				if (!input.includeDocs) {
 					return resolve(definitionInformation);
 				}
-				p = cp.execFile(godoc, [pkgPath], { env }, (err, stdout, stderr) => {
-					if (err && (<any>err).code === 'ENOENT') {
-						vscode.window.showInformationMessage('The "godoc" command is not available.');
-					}
-					let godocLines = stdout.toString().split('\n');
-					let doc = '';
-					let sigName = signature.substring(0, signature.indexOf(' '));
-					let sigParams = signature.substring(signature.indexOf(' func') + 5);
-					let searchSignature = 'func ' + sigName + sigParams;
-					for (let i = 0; i < godocLines.length; i++) {
-						if (godocLines[i] === searchSignature) {
-							while (godocLines[++i].startsWith('    ')) {
-								doc += godocLines[i].substring(4) + '\n';
-							}
-							break;
-						}
-					}
-					if (doc !== '') {
+				runGodoc(pkgPath, input.word, token).then(doc => {
+					if (doc) {
 						definitionInformation.doc = doc;
 					}
-					return resolve(definitionInformation);
+					resolve(definitionInformation);
+				}, () => {
```

Comment:
Shouldn't this be `.catch()` so that it also handles errors in the `then` callback? Are you sure you want to squelch any errors here and not log them anywhere?